ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala

```diff
@@ -60,6 +69,9 @@ class InMemoryLedger(
     l
   }
 
+  // TODO(RC): Do we need a clone here? The package store contains mutable state.
+  private val packages = packages0
```

Comment:
@gaboraranyossy-da @gerolf-da Opinions? In practice, the ownership of the package store is "moved" from the sandbox initialization part (scenario loading) to the `InMemoryLedger` instance.

Reply:
It's not really a clone, just a copied reference. I don't see why we would need it.

Reply:
Yes, this is just a copied reference. I was wondering whether we need a clone instead, given that the reference points to mutable data.

Reply:
we are just reading here, aren't we? In that case I'd define a pure read-concerned interface for this purpose and have it passed as a dependency instead of the whole package store.

Reply:
New packages can be added at run time through the new admin API, see https://github.com/digital-asset/daml/pull/1818/files#diff-f0fa0fae9e9e5750f0db679f20742a95R263

Reply:
oh, right. Are we mutating the packages somewhere else too?

Reply:
No, the package store is initialized as part of the scenario loading, and after that passed to the ledger. It should not be modified from the outside.